components/admin/lessons/AdminLessonExerciseCard.test.js

```diff
@@ -0,0 +1,187 @@
+      variables: { id: 1 }
+    },
+    result: {
+      data: {
+        deleteExercise: {
+          id: 1
+        }
+      }
+    },
+    // Keep the query at loading state for ~3170 years
+    delay: 100_000_000_000_000
+  }
+]
+
+describe('AdminLessonExerciseCard component', () => {
+  it('Should show explanation on click', async () => {
+    expect.assertions(1)
+
+    render(
+      <MockedProvider>
+        <Component user={user} exercise={exercise} />
+      </MockedProvider>
+    )
+
+    const btn = screen.getByText('Show explanation')
+    await userEvent.click(btn)
+
+    expect(screen.getByText('Hide explanation')).toBeTruthy()
+  })
+
+  it('Should remove exercise', async () => {
+    expect.assertions(1)
+
+    const onRemoveMock = jest.fn()
+
+    render(
+      <MockedProvider mocks={mocks}>
+        <Component user={user} exercise={exercise} onRemove={onRemoveMock} />
+      </MockedProvider>
+    )
+
+    const btn = screen.getByText('REMOVE EXERCISE')
+    await userEvent.click(btn)
+
+    expect(onRemoveMock).toBeCalled()
+  })
+
+  it('Should not call onRemove if not passed', async () => {
+    expect.assertions(1)
+
+    const onRemoveMock = jest.fn()
+
+    render(
+      <MockedProvider mocks={mocks}>
+        <Component user={user} exercise={exercise} />
+      </MockedProvider>
+    )
+
+    const btn = screen.getByText('REMOVE EXERCISE')
+    await userEvent.click(btn)
+
+    expect(onRemoveMock).not.toBeCalled()
+  })
+
+  it('Should unflag exercise', async () => {
+    expect.assertions(1)
+
+    const onUnflag = jest.fn()
+
+    render(
+      <MockedProvider mocks={mocks}>
+        <Component user={user} exercise={exercise} onUnflag={onUnflag} />
+      </MockedProvider>
+    )
+
+    const btn = screen.getByText('UNFLAG EXERCISE')
+    await userEvent.click(btn)
+
+    expect(onUnflag).toBeCalled()
+  })
+
+  it('Should not call unflag if not passed', async () => {
+    expect.assertions(1)
+
+    const onUnflag = jest.fn()
+
+    render(
+      <MockedProvider mocks={mocks}>
+        <Component user={user} exercise={exercise} />
+      </MockedProvider>
+    )
+
+    const btn = screen.getByText('UNFLAG EXERCISE')
+    await userEvent.click(btn)
+
+    expect(onUnflag).not.toBeCalled()
+  })
+
+  it('Should show loading spinner', async () => {
+    expect.assertions(1)
+
+    render(
+      <MockedProvider mocks={loadingMocks}>
+        <Component user={user} exercise={exercise} onRemove={() => {}} />
+      </MockedProvider>
+    )
+
+    const btn = screen.getByText('REMOVE EXERCISE')
+    await userEvent.click(btn)
+
+    expect(screen.queryByText('REMOVE EXERCISE')).toBeFalsy()
+  })
+
+  it('Should not show the explanation if not provided', () => {
+    expect.assertions(1)
+
+    render(
+      <MockedProvider mocks={mocks}>
+        <Component user={user} exercise={{ ...exercise, explanation: '' }} />
+      </MockedProvider>
+    )
+
+    expect(screen.queryByText('Show explanation')).toBeFalsy()
```

Comment:
nit: it might to be clearer to test that element does not exist
```javascript
expect(screen.getByText('Show explanation')).not.toBeInTheDocument()
```